libraries/standard/http/utest/http_utest.c

```diff
@@ -59,8 +80,81 @@
 
 #define HTTP_TEST_BUFFER_LEN    1024
 
+/* File-scoped Global variables */
+static HTTPStatus_t retCode = HTTP_NOT_SUPPORTED;
+static uint8_t testBuffer[ HTTP_TEST_BUFFER_SIZE ] = { 0 };
+static HTTPRequestHeaders_t testHeaders = { 0 };
+static _headers_t expectedHeaders = { 0 };
+static int testRangeStart = 0;
+static int testRangeEnd = 0;
 static uint8_t httpBuffer[ HTTP_TEST_BUFFER_LEN ] = { 0 };
 
+/* ============================ Helper Functions ============================== */
+
+/**
+ * @brief Fills the test input buffer and expectation buffers with pre-existing data
+ * before calling the API function under test.
+ */
+static void setupBuffersWithPreexistingHeader( HTTPRequestHeaders_t * testRequestHeaders,
+                                               uint8_t * testBuffer,
+                                               size_t bufferSize,
+                                               _headers_t * expectedHeaders,
+                                               const char * preexistingData )
+{
+    size_t dataLen = strlen( preexistingData );
+
+    testRequestHeaders->pBuffer = testBuffer;
+    testRequestHeaders->bufferLen = bufferSize;
+    int numBytes = snprintf( ( char * ) testRequestHeaders->pBuffer,
+                             bufferSize,
+                             "%s",
+                             preexistingData );
+    /* Make sure that the entire pre-existing data was printed to the buffer. */
+    TEST_ASSERT_GREATER_THAN( 0, numBytes );
+    TEST_ASSERT_LESS_THAN( bufferSize, ( size_t ) numBytes );
+    testRequestHeaders->headersLen = dataLen;
+
+    /* Fill the same data in the expected buffer as HTTPClient_AddRangeHeaders()
+     * is not expected to change it. */
+    TEST_ASSERT( memcpy( expectedHeaders->buffer, testRequestHeaders->pBuffer,
```

Comment:
A redundant check on memcpy that will never return an error code

Reply:
Done